pkg/services/store/entity/health.go

```diff
@@ -0,0 +1,73 @@
+package entity
+
+import (
+	"context"
+
+	grpcAuth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"google.golang.org/grpc/health/grpc_health_v1"
+)
+
+// Compile-time assertion
+var _ HealthService = &healthServer{}
+
+type HealthService interface {
+	grpc_health_v1.HealthServer
+	grpcAuth.ServiceAuthFuncOverride
+}
+
+func ProvideHealthService(server EntityStoreServer) (grpc_health_v1.HealthServer, error) {
+	h := &healthServer{entityServer: server}
+	return h, nil
+}
+
+type healthServer struct {
+	entityServer EntityStoreServer
+}
+
+// AuthFuncOverride for no auth for health service.
+func (s *healthServer) AuthFuncOverride(ctx context.Context, _ string) (context.Context, error) {
+	return ctx, nil
+}
+
+func (s *healthServer) Check(ctx context.Context, req *grpc_health_v1.HealthCheckRequest) (*grpc_health_v1.HealthCheckResponse, error) {
+	h, err := s.entityServer.IsHealthy(ctx, &HealthRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	if h.Healthy {
+		return healthyResponse(), nil
+	} else {
+		return unhealthyResponse(), nil
+	}
+}
+
+func (s *healthServer) Watch(req *grpc_health_v1.HealthCheckRequest, stream grpc_health_v1.Health_WatchServer) error {
+	h, err := s.entityServer.IsHealthy(stream.Context(), &HealthRequest{})
```

Comment:
IIUC, the Watch endpoint should implement [server side streaming](https://grpc.io/docs/languages/go/basics/#server-side-streaming-rpc). I am expecting a for loop here ?

Reply:
Assuming we're trying to implement https://github.com/grpc/grpc/blob/master/doc/health-checking.md it seems like we should update the HealthResponse proto to match the definition in the doc, and that yes we should have a loop to check the health periodically and report back if it changes.

Reply:
`A client can call the Watch method to perform a streaming health-check. The server will immediately send back a message indicating the current serving status. It will then subsequently send a new message whenever the service's serving status changes.`

I missed this! So we just keep checking and only send something when the status changes 👍

Reply:
@DanCech we shouldn't need to add anything to the proto. The health response currently in the proto is just the response for our `IsHealthy` rpc. The health check service implementation already uses the `grpc_health_v1.HealthCheckResponse`.

Reply:
I mean it wouldn't hurt to use the same request/response types in the grpc interface for unified storage vs rolling our own